### PR TITLE
Message content view

### PR DESCRIPTION
**Pull request by evsaev**
**This PR**:
- extracts content from `MessageCell` to `MessageContentView`
- makes use of `MessageContentView` throughout message rethink

### CODE REVIEW

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift [hunk 1]
```diff
@@ -5,331 +5,61 @@
...
-                ),
-                bubbleView.topAnchor.pin(equalTo: reactionsView.centerYAnchor)
-            ]
-        }
-        
-        NSLayoutConstraint.activate(constraintsToActivate)
+    var content: _ChatMessage<ExtraData>? {
+        get { messageContentView.content }
+        set { messageContentView.content = newValue }
     }
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift:22]: Let's not do this. These values should be accessed directly on the view

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift [hunk 2]
```diff
@@ -5,331 +5,61 @@
...
-                    self?.delegate?.didTapOnAttachmentAction($0, $1)
-                }
-            )
+    func setUpLayoutIfNeeded(options: ChatMessageLayoutOptions) {
+        if messageContentView.layoutOptions == nil {
+            let alignment = options.contains(.flipped) ?
+                messageContentView.trailingAnchor.pin(equalTo: contentView.trailingAnchor) :
+                messageContentView.leadingAnchor.pin(equalTo: contentView.leadingAnchor)
+            alignment.isActive = true
         }
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift:35]: To me it feels the container should be responsible for content alignment, not the content view. That's why we set the leading/trailing constraint in a cell. However, the `if options == nil` check doesn't look nice but didn't find a better way to set the constraint just once

Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift:35]: Can we keep all alignment stuff in one place -> inside the content view?

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift [hunk 3]
```diff
@@ -5,331 +5,61 @@
...
+            messageContentView
+                .topAnchor
+                .pin(equalTo: contentView.topAnchor),
+            messageContentView
+                .bottomAnchor
+                .pin(equalTo: contentView.bottomAnchor),
+            messageContentView
+                .widthAnchor
+                .pin(lessThanOrEqualTo: contentView.widthAnchor, multiplier: 0.75)
+        ])
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift:53]: The content should decide about its ideal width.

**Review by VojtaStavik — APPROVED** [review]

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift (hunk 2 above)
Comment (evsaev) [reply Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift]: done

File: Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift (hunk 1 above)
Comment (evsaev) [reply Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift]: fixed 👍

**Pull request merged**